exe/Main.hs

```diff
@@ -2,38 +2,58 @@
 {-# LANGUAGE RecordWildCards #-}
 module Main where
 
+import Control.Concurrent
+import Control.Monad
+import Data.List
+
 import Miso
-import Miso.String
+import Miso.String hiding (replicate)
 
-type Model = Int
+type Model = [Int]
 
 main :: IO ()
-main = startApp App { initialAction = SayHelloWorld, ..}
+main = startApp App { initialAction = NoOp
+                    , ..
+                    }
   where
-    model  = 0
+    model  = [1..10]
     update = updateModel
     view   = viewModel
     events = defaultEvents
-    subs   = []
+    subs   = [ ] -- setNums SetNums ]
+
+setNums :: ([Int] -> action) -> Sub action model
+setNums f _ = \sink -> do
+  void . forkIO . forever $ do
+    threadDelay 100000
+    nums <- sort . nub <$> replicateM 20 getRand
+    sink (f nums)
+
+foreign import javascript unsafe "$r = h$rand() % 10;"
+  getRand :: IO Int
 
 updateModel :: Action -> Model -> Effect Action Model
-updateModel AddOne m = noEff (m + 1)
-updateModel SubtractOne m = noEff (m - 1)
 updateModel NoOp m = noEff m
-updateModel SayHelloWorld m = m <# do
-  putStrLn "Hello World!" >> pure NoOp
+updateModel (SetNums xs) _ = noEff xs
+updateModel (Say x) m = m <# do
+  print x >> pure NoOp
+updateModel NewNums m = m <# do
+  nums <- replicateM 40 getRand
+  print nums >> pure (SetNums nums)
```

Comment:
Fwiw, I don’t think this should replace the counter example. The counter example seems easier to understand and thereby a better choice for the initial example people see. Obviously adding a separate example for this is totally fine.

Reply:
Agreed, I'm going to undo this change.